spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/PropertiesLauncher.java

```diff
@@ -142,48 +140,61 @@ public PropertiesLauncher() {
 	}
 
 	protected File getHomeDirectory() {
-		return new File(SystemPropertyUtils
-				.resolvePlaceholders(System.getProperty(HOME, "${user.dir}")));
+		try {
+			return new File(getPropertyWithDefault(HOME, "${user.dir}"));
+		}
+		catch (Exception ex) {
+			throw new IllegalStateException(ex);
+		}
 	}
 
 	private void initializeProperties() throws Exception, IOException {
-		String config = "classpath:BOOT-INF/classes/"
-				+ SystemPropertyUtils.resolvePlaceholders(
-						SystemPropertyUtils.getProperty(CONFIG_NAME, "application"))
-				+ ".properties";
-		config = SystemPropertyUtils.resolvePlaceholders(
-				SystemPropertyUtils.getProperty(CONFIG_LOCATION, config));
-		InputStream resource = getResource(config);
-		if (resource != null) {
-			log("Found: " + config);
-			try {
-				this.properties.load(resource);
-			}
-			finally {
-				resource.close();
+		List<String> configs = new ArrayList<String>();
+		if (getProperty(CONFIG_LOCATION) != null) {
+			configs.add(getProperty(CONFIG_LOCATION));
+		}
+		else {
+			// TODO: remove support for "application.properties"
+			String[] names = getPropertyWithDefault(CONFIG_NAME, "loader,application")
```

Comment:
Thanks. Perhaps I've misread the change, but won't this warn if your app uses `application.properties` at all? For example, if there's one in `BOOT-INF/classes` in the jar that's configuring properties that don't affect `PropertiesLauncher`? I think we need to warn only if `application.properties` is found and it's used to configure one of the launcher's properties.